### Package imports

In [32]:
import pandas as pd
import numpy as np
#import texthero as hero # MUST BE V1.0.5 (pip install texthero==1.0.5)
import seaborn as sns
import matplotlib.pyplot as plt
from collections import Counter

### Reading in the data

In [7]:
train = pd.read_csv("AfrikaansPOSData/train.csv")


### Making sure data looks correct

In [12]:
train.head(5)

Token  POS
0        Die   LB
1       doel  NSE
2        van  SVS
3        die   LB
4  webtuiste  NSE

In [2]:
train.tail(5)

NameError: name 'train' is not defined

Initial checks indicate the presence of missing values. Looking at the dataframe, we see there are NaN rows after each sentence ends.

In [12]:
train['Token'].isna().sum() 

2613

We have 2613 NaN rows in training data. We can also infer that these are many how sentences are contained in the dataset.


In [23]:
train.dropna(inplace=True) # Dropping NaNs

In [ ]:
train.shape

ValueError: too many values to unpack (expected 2)

In [33]:
# check total words in vocabulary
wordCounts = Counter(train.Token)
wordCounts = np.argsort(wordCounts)[::-1]

tagCounts = Counter(train.POS)

In [34]:
counts

Counter({'Die': 368,
         'doel': 21,
         'van': 2146,
         'die': 4222,
         'webtuiste': 25,
         'vir': 573,
         'Suid-Afrikaanse': 82,
         'Regeringsdienste': 1,
         'is': 987,
         'om': 973,
         "'n": 1218,
         'enkele': 5,
         'bron': 3,
         'inligting': 120,
         'te': 1150,
         'bied': 29,
         'oor': 192,
         'dienste': 56,
         'wat': 1002,
         'deur': 299,
         'regering': 43,
         'verskaf': 55,
         'word': 720,
         '.': 2084,
         'oogmerk': 1,
         'dispariteit': 1,
         'verminder': 16,
         'tussen': 38,
         'bemagtigdes': 1,
         'en': 1811,
         'ontmagtigdes': 1,
         ',': 2323,
         'oftewel': 1,
         'eerste': 41,
         'tweede': 17,
         'ekonomieë': 2,
         'bestaan': 18,
         'kritiese': 14,
         'ingryp': 1,
         'daarop': 14,
         'gerig': 16,
         'seker': 21,
         'maak': 87,
   